src/cogent3/app/io.py

```diff
@@ -351,7 +358,7 @@ def main(self, path: IdentifierType) -> TabularType:
 
 
 @define_app(app_type=WRITER)
-class write_tabular:
+class write_tabular(_checkpointable):
```

Comment:
Here we mixed decorator approach and inheritance. Do we expect other users to use `_checkpointable` by inheritance or will we change it in the future?

Reply:
change in future